api/fixtures_test.go

```diff
@@ -0,0 +1,1139 @@
+	2. Query the test-internal `proverRoutes` struct for a route-compatible `prover`
+	3. Employ the `prover` to parse the indexer's response into a route appropriate `witness` object
+C. Save `xyzLiveFixture` into a non-git-committed fixture `./test_resources/_FIXTURE_NAME.json` (NOTICE THE `_` PREFIX)
+D. Assert that the non-git-commited fixture equals the git-committed version `./test_resources/FIXTURE_NAME.json`
+
+It is the responsibility of the test writer to check that the generated fixture represent the expected results.
+
+----
+TODO: SHOULD this be used to auto-generate unit test cases in the SDK's?
+In particular, one could craft a generic cucumber test that looks something like:
+
+* ("fixture parsing")
+	Given an indexer fixture file <file-name>.
+* ("fixture mock setup")
+	When a mock indexer server has been set up using the urls and responses provided by the indexer fixture,
+* ("fixture validation")
+	Then iterate through all of the fixture's test cases: query the mock indexer server, parse the response into an SDK object, and validate that it comports with the witness.
+
+The third step ("fixture validation") is unusually high-level for our cucumber tests, but it would allow for a more streamlined unit tests bootstrapping.
+Implementoing the "fixture validation" step would logically be broken down as follows:
+
+For each `testCase`` in the parsed fixture:
+	1. Use the `witness` type together with the `request` to find an appropriate calling SDK `method`.
+	2. Call the SDK `method` against the mock indexer server.
+	3. Validate that the resulting SDK object comports with the `witness`.
+************************************** */
+
+const fixtestListenAddr = "localhost:8999"
+const fixtestBaseURL = "http://" + fixtestListenAddr
+const fixtestMaxStartup time.Duration = 5 * time.Second
+const fixturesDirectory = "test_resources/"
+
+type fixture struct {
+	File         string     `json:"file"`
+	Owner        string     `json:"owner"`
+	LastModified string     `json:"lastModified"`
+	Frozen       bool       `json:"frozen"`
+	Cases        []testCase `json:"cases"`
+}
+type testCase struct {
+	Name         string       `json:"name"`
+	Request      requestInfo  `json:"request"`
+	Response     responseInfo `json:"response"`
+	Witness      interface{}  `json:"witness"`
+	WitnessError *string      `json:"witnessError"`
+}
+type requestInfo struct {
+	Path   string  `json:"path"`
+	Params []param `json:"params"`
+	URL    string  `json:"url"`
+	Route  string  `json:"route"` // `Route` stores the simulated route found in `proverRoutes`
+}
+type param struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+type responseInfo struct {
+	StatusCode int    `json:"statusCode"`
+	Body       string `json:"body"`
+}
+type prover func(responseInfo) (interface{}, *string)
+
+// ---- BEGIN provers / witness generators ---- //
+
+func accountsProof(resp responseInfo) (wit interface{}, errStr *string) {
```

Comment:
I believe these `xyzProof()` functions could be unified with go 1.18 generics. Maybe there's another way?